### PR TITLE
fix(tile-converter): memory overflow

### PR DESCRIPTION
**Pull request by belom88**

### CODE REVIEW

File: modules/tile-converter/src/lib/utils/write-queue.ts [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import {Queue} from './queue';
+import process from 'process';
+
+const MEMORY_LIMIT = 4 * 1024 * 1024 * 1024; // 4Gb
```
Comment (ibgreen) [inline modules/tile-converter/src/lib/utils/write-queue.ts:4]: Maybe a short comment on why this is a good limit?

Also is it gigabit (Gb) or gigabyte (GB)>

File: modules/tile-converter/src/lib/utils/write-queue.ts [hunk 2]
```diff
@@ -18,6 +21,13 @@ export default class WriteQueue<T extends WriteQueueItem> extends Queue<T> {
     this.writeConcurrency = writeConcurrency;
   }
 
+  async enqueue(val: T) {
+    super.enqueue(val);
+    if (process.memoryUsage().rss > MEMORY_LIMIT) {
```
Comment (ibgreen) [inline modules/tile-converter/src/lib/utils/write-queue.ts:26]: I would add some comments and possibly links to relevant Node.js docs here.

**Review by ibgreen — APPROVED** [review]

File: modules/tile-converter/src/lib/utils/write-queue.ts (hunk 1 above)
Comment (belom88) [reply modules/tile-converter/src/lib/utils/write-queue.ts:4]: I'm not sure it is so good. We tested it and it worked well.

Comment (ibgreen) [reply modules/tile-converter/src/lib/utils/write-queue.ts:4]: Then add a comment to say that this is based on testing.

Comment (belom88) [reply modules/tile-converter/src/lib/utils/write-queue.ts]: Done

File: modules/tile-converter/src/lib/utils/write-queue.ts (hunk 2 above)
Comment (belom88) [reply modules/tile-converter/src/lib/utils/write-queue.ts:28]: Done

**Pull request merged**